napari/components/viewer_model.py

```diff
@@ -80,13 +80,13 @@ def __init__(
         self._palette = None
         self.theme = 'dark'
 
+        # These connections will be removed once the EVH reaches the Dims
         self.dims.events.camera.connect(self.reset_view)
-        self.dims.events.ndisplay.connect(self._update_layers)
-        self.dims.events.order.connect(self._update_layers)
-        self.dims.events.axis.connect(self._update_layers)
-        self.layers.events.changed.connect(self._update_active_layer)
-        self.layers.events.changed.connect(self._update_grid)
-        self.layers.events.changed.connect(self._on_layers_change)
+        self.dims.events.ndisplay.connect(lambda e: self._update_layers())
+        self.dims.events.order.connect(lambda e: self._update_layers())
+        self.dims.events.axis.connect(lambda e: self._update_layers())
+
+        self.layers.event_handler.register_listener(self)
```

Comment:
no grid mode is failing because we have a callback for `_on_translate_grid_change` but we never emit a `translate_grid` event!! Hard for it to work without that!!!